docs/samples/box/shadow.md

```diff
@@ -0,0 +1,169 @@
+// </block:setup>
+
+// <block:annotation1:1>
+const annotation1 = {
+  type: 'box',
+  backgroundColor: 'rgba(0,150,0,0.02)',
+  borderColor: 'rgb(0,150,0)',
+  borderRadius: {
+    topLeft: 4,
+    topRight: 4,
+    bottomLeft: 4,
+    bottomRight: 4
+  },
+  borderWidth: 1,
+  xMax: (ctx) => max(ctx, 0, 'x') + 2,
+  xMin: (ctx) => min(ctx, 0, 'x') - 2,
+  yMax: (ctx) => max(ctx, 0, 'y') + 2,
+  yMin: (ctx) => min(ctx, 0, 'y') - 2,
+  beforeDraw: (ctx) => drawShadow(ctx)
+};
+// </block:annotation1>
+
+// <block:annotation2:2>
+const annotation2 = {
+  type: 'box',
+  backgroundColor: 'rgba(150,0,0,0.02)',
+  borderColor: 'rgb(150,0,0)',
+  borderRadius: {
+    topLeft: 4,
+    topRight: 4,
+    bottomLeft: 4,
+    bottomRight: 4
+  },
+  borderWidth: 1,
+  xMax: (ctx) => max(ctx, 1, 'x') + 2,
+  xMin: (ctx) => min(ctx, 1, 'x') - 2,
+  yMax: (ctx) => max(ctx, 1, 'y') + 2,
+  yMin: (ctx) => min(ctx, 1, 'y') - 2,
+  beforeDraw: (ctx) => drawShadow(ctx)
+};
+// </block:annotation2>
+
+/* <block:config:0> */
+const config = {
+  type: 'scatter',
+  data,
+  options: {
+    plugins: {
+      annotation: {
+        annotations: {
+          annotation1,
+          annotation2
+        }
+      }
+    }
+  }
+};
+/* </block:config> */
+
+// <block:utils:3>
+const PI = Math.PI;
+const HALF_PI = PI / 2;
+
+function min(ctx, datasetIndex, prop) {
+  const dataset = ctx.chart.data.datasets[datasetIndex];
+  return dataset.data.reduce((v, point) => Math.min(point[prop], v), Infinity);
+}
+
+function max(ctx, datasetIndex, prop) {
+  const dataset = ctx.chart.data.datasets[datasetIndex];
+  return dataset.data.reduce((v, point) => Math.max(point[prop], v), -Infinity);
+}
+
+function addRoundedRectPath(ctx, rect) {
+  const {x, y, w, h, radius} = rect;
+  ctx.arc(x + radius.topLeft, y + radius.topLeft, radius.topLeft,
+    -HALF_PI, PI, true);
+  ctx.lineTo(x, y + h - radius.bottomLeft);
+  ctx.arc(x + radius.bottomLeft, y + h - radius.bottomLeft, radius.bottomLeft,
+    PI, HALF_PI, true);
+  ctx.lineTo(x + w - radius.bottomRight, y + h);
+  ctx.arc(x + w - radius.bottomRight, y + h - radius.bottomRight, radius.bottomRight,
+    HALF_PI, 0, true);
+  ctx.lineTo(x + w, y + radius.topRight);
+  ctx.arc(x + w - radius.topRight, y + radius.topRight, radius.topRight,
+    0, -HALF_PI, true);
+  ctx.lineTo(x + radius.topLeft, y);
+}
+
+function drawShadow(context) {
+  const ctx = context.chart.ctx;
+  const {x, y, width, height, options} = context.element;
+  ctx.save();
+  ctx.fillStyle = 'transparent';
+  ctx.strokeStyle = options.borderColor;
+  ctx.beginPath();
+  addRoundedRectPath(ctx, {
+    x, y, w: width, h: height,
+    radius: options.borderRadius
+  });
+  ctx.closePath();
+  ctx.fill();
+  ctx.globalCompositeOperation = 'destination-over';
+  ctx.shadowColor = options.borderColor;
+  ctx.shadowBlur = 9;
+  ctx.shadowOffsetX = 5;
+  ctx.shadowOffsetY = -5;
+  ctx.stroke();
+  ctx.restore();
+  return true;
+}
+// </block:utils>
+
+var actions = [
```

Comment:
Using const for other vars, so would be better to use it here too

Reply:
@LeeLenaleee agree! I see that var is iused in many other samples. I think they are coming from previous version.
I fix this new ones and I'm going to submit a PR for the others